lib/alloc.js

```diff
@@ -0,0 +1,40 @@
+class CodeAllocator {
+  constructor (sliceSize) {
+    this.sliceSize = sliceSize;
+    this.slicesPerPage = Process.pageSize / sliceSize;
+
+    this.pages = [];
+    this.free = [];
+  }
+
+  allocateSlice () {
+    let slice = this.free.pop();
+    if (slice !== undefined) {
+      return slice;
+    }
+
+    return this._allocatePage();
+  }
+
+  _allocatePage () {
+    const page = Memory.alloc(Process.pageSize);
+    for (let i = 1; i < this.slicesPerPage; i++) {
+      const slice = page.add(this.sliceSize * i);
```

Comment:
Would be good to avoid re-reading `this.slicesPerPage` and `this.sliceSize` here, as the JS runtime isn't necessarily smart enough to optimize these property reads.